### DISCUSSION TITLE
How about RISC-V support?

### DISCUSSION THREAD

**Discussion by rlanders** [mbed-ce/mbed-os#303] [Ideas]
Before you shoot that down, realize that many new SoC's are coming out that also have a RISC-V mcu in them.

**Comment by multiplemonomials** [comment]
This would be pretty cool, and there's a hell of a lot of Mbed OS that could be reused for RISC-V support!  I would be pretty jazzed about a porting effort to this architecture.

However, two things will become problems pretty quickly.  First of all, Mbed has a fair amount of utility code that's specifically written for the ARM architecture -- stuff like the `mstd_atomic` header comes to mind.  Perhaps a lot of this stuff can be done through the standard library now, but there would need to be a lot of work up front to migrate away from this ARM specific stuff.

Another issue would be the use of the Keil RTX5 RTOS, which is ARM only.  We'd probably have to switch this out for something like FreeRTOS.  I'd actually really support that effort, because it would bring way better debugger compatibility, and it would not be an impossible effort because both RTX5 and FreeRTOS support the CMSIS-RTOS API, so a good chunk of Mbed would not need modifications.  However, it would still take a heck of a lot of work to make it happen.
